### PR TITLE
Pass the event to the click handlers

### PR DESCRIPTION
**Pull request by pashadia**
This would be quite useful, as it would open possibilities for handling actions like Control-Click or Shift-Click.

### CODE REVIEW

**Comment by techniq** [comment]
@pashadia I agree this would be useful, although I don't like passing it as the second argument (as events are typically first).  This is an oversight on my part though.

I'd be willing to switch the order (event, then columns/data) and release it as a semver major version.

Also, for columns/data, in `mui-table` we now pass these as a single [object](https://github.com/techniq/mui-table/blob/master/src/index.js#L289) so you can de-structure what you need (sometimes you just need the data, sometimes just the columns, so this made it more convenient).  Should we embrace this change as well if we do make the breaking change?

**Comment by pashadia** [comment]
I agree, `event` tends to be the first parameter. I didn't want to introduce a breaking change before discussing, as they tend to not be accepted :)

I've just started using `mui-virtualized-table`, so I don't mind breaking changes at all.

**Comment by pashadia** [comment]
@techniq Let me know if there's anything I could do to help.

**Comment by techniq** [comment]
If you could update the PR to pass event as the first argument and the
second argument as an object of rowData/columns/etc like mui-table along
with the related storybooks, docs, I'll try to get it merged quickly and
cut a major semver release.

**Comment by pashadia** [comment]
Made the changes, along with:

1. Passing the full data param, since it could be useful for extra-selects (like Shift-Click selecting)
2. Adding double click and context menu handlers.

File: src/index.js [hunk 1]
```diff
@@ -301,10 +311,10 @@ class MuiTable extends Component {
             }
             style={{ width: 'inherit' }} // fix text overflowing
             direction={orderDirection}
-            onClick={() =>
+            onClick={(event) =>
               column.onHeaderClick
-                ? column.onHeaderClick()
-                : onHeaderClick(column)
+                ? column.onHeaderClick(event)
+                : onHeaderClick(column, event)
```
Comment (techniq) [inline src/index.js]: This should pass `event` as the first argument, and `column` within an object.

```suggestion
                : onHeaderClick(event, { column })
```

**Comment by techniq** [comment]
@pashadia Thanks for the update.  I requested a change to your PR, but along with that, the following will need updated as well:

- Update the storybook [stories](https://github.com/techniq/mui-virtualized-table/blob/master/stories/basic.js) to reflect this change.
- Update the Typescript [definitions](https://github.com/techniq/mui-virtualized-table/blob/master/index.d.ts#L33).

**Comment by pashadia** [comment]
Thanks.

I'll update the stories asap, however I'll need some help with the
typescript definitions.

**Comment by techniq** [comment]
@pashadia No worries, I'll update the TypeScript definitions

**Comment by techniq** [comment]
@pashadia Sorry for the big delay, I merged this (with a small adjustment to use `isCellDisabled()` instead of `rowData.disabled`).  You should be able to use something like this:

```jsx
isCellDisabled={({ rowData }) => rowData.disabled }
```

I still need to get the Typescript definitions updated and do some more testing before cutting a release.  @cvanem could you take a look at getting this ready for release?  Thanks

**Comment by cvanem** [comment]
@techniq @pashadia 

Sure I will take a look.  I noticed this now passes the full data set back with the onClick events.  Are we sure we want to do this?  Wouldn't it make more sense for the data state to be accessed externally?
 
How does this look for the onClick event handler types?

```
export type MouseClickEventHandler<HTMLElement> = (
  event: React.MouseEvent<HTMLElement>,
  props: {
    column: IMuiVirtualizedTableColumn<TRow>;
    rowData: TRow;
    data: TRow[];
  }
) => void;

onCellClick?: MouseClickEventHandler,
onCellDoubleClick?: MouseClickEventHandler,
onCellContextMenu?: MouseClickEventHandler,  

```

**Comment by techniq** [comment]
@cvanem Thanks for the PR and sorry for the delay on the merge, I've been out of town this week.

I'm indifferent either way with passing the full data in the events... sometimes with composition / usage of a component you don't always have access the props (or can expose hooks that expose props/state helpers that wouldn't know directly about the component).

For example, I've created a `useTableOrder`, `useTablePagination`, `useTableSelection`, and `useTableHierarchy` hooks/render prop components that expose these type of prop/state helpers.

With that said, I don't specifically have a use case where I need the full data, so right now, I'm indifferent to adding them at this time.

Also, since this is a breaking change, should we go ahead and upgrade to Material-UI 4.0 like you mentioned in #25?

**Comment by cvanem** [comment]
@techniq For passing the full data set on events, I am mainly worried about performance issues with passing large datasets around.  Maybe we should run some tests and see if it causes any slowdown?  I am also OK with leaving it in and just wait and see.

As for MUI 4, yes it is a good time to update.  I'm pretty busy right now but could probably do this late next week.

**Comment by techniq** [comment]
@pashadia This is released as `next` / [3.0.0-0](https://github.com/techniq/mui-virtualized-table/releases/tag/v3.0.0-0) if you want to give it a test before releasing as stable.

**Comment by pashadia** [comment]
@techniq This is great, thank you! 
Unfortunately, I'm not sure I can find the time in the near future, and I don't want to slow things down.

**Comment by techniq** [comment]
@pashadia No worries. I'll be without a laptop next week (vacation, and getting my Macbook keyboard replaced) but it will continue to be available using `npm install mui-virtualized-table@next`).  It can even be a few weeks before we push to stable.

**Pull request merged**